Core/include/Acts/Vertexing/SeedVertexFinder.ipp

```diff
@@ -0,0 +1,656 @@
+
+  // plane (alpha*x + beta*y + gamma*z + delta = 0), splitted to {{alpha, beta,
+  // gamma}, delta} for convenience
+  return {abg, delta};
+}
+
+template <typename spacepoint_t>
+Acts::Vector3 Acts::SeedVertexFinder<spacepoint_t>::findClosestPointFromPlanes(
+    const std::vector<Acts::SeedVertexFinder<spacepoint_t>::Triplet>& triplets)
+    const {
+  // define function f = sum over all triplets [distance from an unknown point
+  // (x_0,y_0,z_0) to the plane defined by the triplet] find minimum of "f" by
+  // partial derivations over x_0, y_0, and z_0 each derivation has parts
+  // lineary depending on x_0, y_0, and z_0 (will fill A[deriv][3]) or to
+  // nothing (will fill B[deriv]) solve A*(x_0,y_0,z_0) = B
+
+  Acts::Vector3 vtx = Acts::Vector3::Zero();
+  Acts::Vector3 vtx_prev{m_cfg.rMaxFar, m_cfg.rMaxFar, m_cfg.maxAbsZ};
+
+  // (alpha-beta-gamma, delta), distance
+  std::vector<
+      std::pair<std::pair<Acts::Vector3, Acts::ActsScalar>, Acts::ActsScalar>>
+      triplets_with_planes;
+  triplets_with_planes.reserve(triplets.size());
+
+  for (const auto& triplet : triplets) {
+    auto abgd = makePlaneFromTriplet(triplet);
+    triplets_with_planes.emplace_back(abgd, -1.);
+  }
+
+  // elements of the linear equations to solve
+  Acts::SymMatrix3 A = Acts::SymMatrix3::Zero();
+  Acts::Vector3 B = Acts::Vector3::Zero();
+  for (const auto& triplet : triplets_with_planes) {
+    const auto& abg = triplet.first.first;
+    const auto& delta = triplet.first.second;
+
+    A += 2. * (abg * abg.transpose());
+    B -= 2. * delta * abg;
+  }
+
+  for (int iter = 0; iter <= m_cfg.maxIterations; iter++) {
+    // new vertex position
+    vtx = A.lu().solve(B);
+
+    Acts::Vector3 vtx_diff = vtx - vtx_prev;
+
+    if (std::sqrt(vtx_diff.dot(vtx_diff)) < m_cfg.minVtxShift) {
+      // difference between the new vertex and the old vertex is not so large
+      break;
+    }
+
+    if (iter != m_cfg.maxIterations) {
+      // is not the last iteration
+      vtx_prev = vtx;
+
+      for (auto& triplet : triplets_with_planes) {
+        const auto& abg = triplet.first.first;
+        const auto& delta = triplet.first.second;
+        Acts::ActsScalar distance = fabs(abg.dot(vtx) + delta);
+
+        triplet.second = distance;
+      }
+
+      std::sort(triplets_with_planes.begin(), triplets_with_planes.end(),
+                [](const auto& lhs, const auto& rhs) {
+                  return lhs.second < rhs.second;
+                });
+
+      unsigned int threshold = (unsigned int)(triplets_with_planes.size() *
+                                              (1. - m_cfg.removeFraction));
+
+      for (unsigned int tr = threshold + 1; tr < triplets_with_planes.size();
+           ++tr) {
+        const auto& abg = triplets_with_planes.at(tr).first.first;
+        const auto& delta = triplets_with_planes.at(tr).first.second;
+
+        // remove this triplet from A and B
+        A -= 2. * (abg * abg.transpose());
```

Comment:
`squaredNorm`?

Reply:
Hi @andiwand, same as above, we'd like to make a matrix.

Cheers, Petr